templates/CRM/Contact/Page/View/GroupContact.tpl

```diff
@@ -125,24 +125,42 @@
           <th>{ts}Status{/ts}</th>
           <th>{ts}Date Added{/ts}</th>
           <th>{ts}Date Removed{/ts}</th>
+          <th>{ts}Actions{/ts} {help id='actions' file='CRM/Contact/Page/View/GroupContact.hlp'}</th>
           <th></th>
         </tr>
         </thead>
         {foreach from=$groupOut item=row}
           <tr id="group_contact-{$row.id}" class="crm-entity {cycle values="odd-row,even-row"}">
             <td class="bold">
               <a href="{crmURL p='civicrm/group/search' q="reset=1&force=1&context=smog&gid=`$row.group_id`"}">
-                {$row.title}
+                {if $row.saved_search_id}* {/if}{$row.title}
               </a>
             </td>
             <td class="status-removed">{ts 1=$row.out_method}Removed (by %1){/ts}</td>
             <td data-order="{$row.date_added}">{$row.date_added|crmDate}</td>
             <td data-order="{$row.out_date}">{$row.out_date|crmDate}</td>
-            <td>{if $permission EQ 'edit'}
+            <td>
+              {if $permission EQ 'edit'}
+                {if $row.saved_search_id}
+                <a class="action-item crm-hover-button" href="#Added" title="{ts 1=$displayName 2=$row.title}Add %1 manually into %2, overriding smart group critera?{/ts}">
+                  {ts}Manual Add{/ts}
+                {else}
                 <a class="action-item crm-hover-button" href="#Added" title="{ts 1=$displayName 2=$row.title}Add %1 back into %2?{/ts}">
-                  {ts}Rejoin Group{/ts}</a>
-              <a class="action-item crm-hover-button" href="#Deleted" title="{ts 1=$displayName 2=$row.title}Delete %1 from %2? (this group will no longer be listed under Past Groups).{/ts}">
-                {ts}Delete{/ts}</a>{/if}
+                  {ts}Rejoin Group{/ts}
```

Comment:
This needs a </a> on the end (note you open the a tag on New Line 148)

Reply:
Yes, this was poorly done, sorry. Now should be better.